x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/source_fields_merging/utils/build_fields_key_as_array_map.test.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { buildFieldsKeyAsArrayMap } from './build_fields_key_as_array_map';
+
+describe('buildFieldsKeyAsArrayMap()', () => {
+  it('returns primitive type if it passed as source', () => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    expect(buildFieldsKeyAsArrayMap(1 as any)).toEqual({});
```

Comment:
A ts-expect-error might be a little cleaner than this and the accompanying type cast, 🤷

```suggestion
    // @ts-expect-error
    expect(buildFieldsKeyAsArrayMap(1).toEqual({});
```

Reply:
agree, thanks for the suggestion